kolibri/core/content/api.py

```diff
@@ -1199,6 +1200,41 @@ def list(self, request, **kwargs):
         )
 
 
+class BookmarkFilter(FilterSet):
+    available = BooleanFilter(
+        method="filter_available",
+    )
+    kind = CharFilter(
+        method="filter_kind",
+    )
+
+    class Meta:
+        model = Bookmark
+        fields = ("kind",)
+
+    def filter_kind(self, queryset, name, value):
+        queryset = queryset.annotate(
+            kind=Subquery(
+                models.ContentNode.objects.filter(
+                    available=True,
```

Comment:
Can probably leave out filtering by `available` in the subquery, as that will give null values for non-available nodes here.